external/java/src/main/java/ch/epfl/dedis/calypso/CalypsoRPC.java

```diff
@@ -0,0 +1,164 @@
+            return new Ed25519Point(reply.getX());
+        } catch (InvalidProtocolBufferException e) {
+            throw new CothorityCommunicationException(e);
+        }
+    }
+
+    /**
+     * Create a long-term-secret (LTS) and retrieve its configuration.
+     *
+     * @return The LTS configuration that is needed to execute the write contract.
+     * @throws CothorityCommunicationException
+     */
+    public LTS createLTS() throws CothorityCommunicationException, CothorityCryptoException {
+        Calypso.CreateLTS.Builder b = Calypso.CreateLTS.newBuilder();
+        b.setRoster(getRoster().toProto());
+        b.setBcid(getGenesisBlock().getId().toProto());
+
+        ByteString msg = getRoster().sendMessage("Calypso/CreateLTS", b.build());
+
+        try {
+            Calypso.CreateLTSReply resp = Calypso.CreateLTSReply.parseFrom(msg);
+            return new LTS(resp);
+        } catch (InvalidProtocolBufferException e) {
+            throw new CothorityCommunicationException(e);
+        }
+    }
+
+    /**
+     * Ask the secret-manageemnt cothority for the decryption shares.
+     *
+     * @param writeProof The proof of the write request.
+     * @param readProof  The proof of the read request.
+     * @return All the decryption shares that can be used to reconstruct the decryption key.
+     * @throws CothorityCommunicationException
+     */
+    public DecryptKeyReply tryDecrypt(Proof writeProof, Proof readProof) throws CothorityCommunicationException {
+        Calypso.DecryptKey.Builder b = Calypso.DecryptKey.newBuilder();
+        b.setRead(readProof.toProto());
+        b.setWrite(writeProof.toProto());
+
+        ByteString msg = getRoster().sendMessage("Calypso/DecryptKey", b.build());
+
+        try {
+            Calypso.DecryptKeyReply resp = Calypso.DecryptKeyReply.parseFrom(msg);
+            return new DecryptKeyReply(resp);
+        } catch (InvalidProtocolBufferException e) {
+            throw new CothorityCommunicationException(e);
+        }
+    }
+
+    /**
+     * @return the id of the Long Term Secret
+     */
+    public LTSId getLTSId() {
+        return lts.getLtsId();
+    }
+
+    /**
+     * @return the shared public key of the Long Term Secret
+     */
+    public Point getLTSX() {
+        return lts.getX();
+    }
+
+    /**
+     * @return the Long Term Secret.
+     */
+    public LTS getLTS(){
+        return lts;
+    }
+
+    /**
+     * Connects to an existing byzcoin and an existing Long Term Secret.
+     *
+     * @param roster    the nodes handling the byzcoin ledger
+     * @param byzcoinId the id of the byzcoin ledger to connect to
+     * @param ltsId     the id of the Long Term Secret to use
+     * @return CalypsoRPC if everything was found
+     * @throws CothorityException
+     */
+    public static CalypsoRPC fromCalypso(Roster roster, SkipblockId byzcoinId, LTSId ltsId) throws CothorityException {
```

Comment:
if we change how byzcoin constructors work (use `fromExisting` to connect to existing services), we should also change this.

Reply:
I suppose you're referencing the `ByzCoinRPC` constructor.